src/python/foglamp/core/server_daemon.py

```diff
@@ -75,38 +83,54 @@ def start():
             _start_server()
 
 
-def stop():
-    """Stops the FogLAMP process if it is running"""
+class StopProcessFailed(Exception):
+    pass
 
-    pid = get_pid()
 
-    if pid is None:
+def stop(pid = None):
+    """Stops FogLAMP if it is running
+
+    :param pid: Optional pid of the daemon process
+    """
+
+    if not pid:
```

Comment:
There is a popular idea among the python community to threat none and false together and not specifically check for none unless there is a difference between none and 0. pid 0 is not valid.